fetch_printer_status.py

```diff
@@ -0,0 +1,26 @@
+import json, ssl
+
+import paho.mqtt.subscribe as subscribe
+from data.printer_config import PRINTERS
+
+def get_printer_status():
+
+    for p in PRINTERS:
+        printer = PRINTERS[p]
+
+        auth = {"username":printer["username"],"password":printer["access_code"]}
+        tls = ssl._create_unverified_context()
+
+        msg = subscribe.simple(topics=printer["topic_name"],hostname=printer["ip"],port=printer["port"],auth=auth,tls=tls)
+        printer_object = json.loads(msg.payload)
+
+        name = printer["name"]
+        job = printer_object["print"]["subtask_name"]
+        state = printer_object["print"]["gcode_state"]
+        mins = printer_object["print"]["mc_remaining_time"]
```

Comment:
this is the same logic as `get_status_from_mqtt()`.  I'd just include `import archive_retrieve` and then call that function here instead.  It'll make this a nice helper file with just a few lines of code!

Reply:
I got you!! tested and ret tah go

```python
import json
import fdmprinting.archive_retrieve as ar
from data.printer_config import PRINTERS
def get_printer_status():
    print("Fetching JSON for printers")
    for printer_id in PRINTERS:
        printer = PRINTERS[printer_id]
        with open("./"+ printer["name"] + ".json", "w") as my_file:
            print("Writing JSON for", printer["name"])
            my_file.write(json.dumps(ar.get_status_from_mqtt(printer, printer_id), indent=4))
    print("Done!")

if __name__ == "__main__":
    get_printer_status()
```

Reply:
ah - I see my function is pretty naive and misses a lot of your nice formatting :shrug:

Reply:
oh yeah! I hadn't even thought of that - but that is exactly why I bothered to save the raw JSON, so we could easily operate on the data. I hadn't considered the possibility of not needing to add extra columns - that'd be a nice benefit!

Maybe we pass raw JSON field to `get_status_msg()` and have it extract if it's in an error state?  I'm happy to try and figure that - shouldn't be too bad, yah?